### PR TITLE
[Serializer] Allow filtering "object" when using "getSupportedTypes()"

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Follows #49291

This allows telling when only objects are accepted, to skip calling `supportsNormalization()` when native types are considered.

/cc @tucksaun

### CODE REVIEW

File: src/Symfony/Component/Serializer/Normalizer/NormalizerInterface.php [hunk 1]
```diff
@@ -63,10 +58,13 @@ public function supportsNormalization(mixed $data, string $format = null /* , ar
...
      * (a result cannot be cached when it depends on the context or on the data.)
+     * A null value means that the normalizer does not support the corresponding
+     * type.
      *
-     * The special type '*' can be used to indicate that the normalizer might
-     * support any types. A null value means that the normalizer does not support
-     * the corresponding type.
+     * There are 3 special types:
+     *  - "object" indicates that the normalizer might support any classes;
+     *  - "native-*" indicates that the normalizer might support any native types;
```
Comment (stof) [inline src/Symfony/Component/Serializer/Normalizer/NormalizerInterface.php]: this documentation misses the explanation of the `native-` prefix for non-object types (`native-bool`, `native-int` and so one), which is what this `native-*` covers.

This concept of `native types` is not that clear IMO given that this is not a naming used by PHP itself (they talk about "builtin types")

Comment (nicolas-grekas) [reply src/Symfony/Component/Serializer/Normalizer/NormalizerInterface.php]: Good call. Your comment made me realize that we don't need the `native-*` wildcard; `object` is enough.

File: src/Symfony/Component/Serializer/Normalizer/DenormalizerInterface.php [hunk 2]
```diff
@@ -72,10 +67,11 @@ public function supportsDenormalization(mixed $data, string $type, string $forma
...
+     * A null value means that the denormalizer does not support the corresponding
+     * type.
      *
-     * The special type '*' can be used to indicate that the denormalizer might
-     * support any types. A null value means that the denormalizer does not support
-     * the corresponding type.
+     * Use type "object" to match any classes or interfaces,
+     * and type "*" to match any types.
      *
      * @return array<class-string|'*'|string, bool|null>
```
Comment (tucksaun) [inline src/Symfony/Component/Serializer/Normalizer/DenormalizerInterface.php]: as we specified '*', should we also specify 'object' to document the special case ?
```suggestion
     * @return array<class-string|'*'|'object'|string, bool|null>
```

**Review by tucksaun — COMMENTED** [review]
I like the approach as it allows us to be more precise for generic (de)normalizers!
I will try to bench this PR tonight on my (massive) use case.

**Review by tucksaun — APPROVED** [review]

**Comment by tucksaun** [comment]
Bench results: no overall changes, still ~80% gain on `getNormalizer` and 15% gain on the main WT

**Pull request merged**